client/src/periodUtils.js

```diff
@@ -31,17 +35,28 @@ export const PERIOD_FACTORIES = {
     start: date.clone().subtract(offset, "weeks").startOf("week"),
     end: date.clone().subtract(offset, "weeks").endOf("week")
   }),
-  // FIXME: biweekly calculation should be changed, first agree on what it means
-  [RECURRENCE_TYPE.BIWEEKLY]: (date, offset) => ({
-    start: date
-      .clone()
-      .subtract(2 * offset, "weeks")
-      .startOf("week"),
-    end: date
+  [RECURRENCE_TYPE.BIWEEKLY]: (date, offset) => {
+    // every biweekly period's start is even number of weeks apart from reference monday
+    const refMonday = moment(refMondayForBiweekly).startOf(weekType)
+    const curWeekMonday = date.startOf(weekType)
+
+    const diffInWeeks = refMonday.diff(curWeekMonday, "weeks")
+    // current biweekly period's start has to be even number of weeks apart from reference monday
+    const curBiweeklyStart =
+      diffInWeeks % 2 === 0
+        ? curWeekMonday
+        : curWeekMonday.clone().subtract(1, "weeks")
+
+    const curBiweeklyEnd = curBiweeklyStart
       .clone()
-      .subtract(2 * offset, "weeks")
-      .endOf("week")
-  }),
+      .endOf(weekType)
+      .add(1, "weeks")
+
+    return {
+      start: curBiweeklyStart.clone().subtract(2 * offset, "weeks"),
+      end: curBiweeklyEnd.clone().subtract(2 * offset, "weeks")
```

Comment:
```suggestion
      end: curBiweeklyEnd.clone().subtract(2 * offset, "weeks").endOf("day")
```

Reply:
We already do the endOf operation above:
```
curBiweeklyStart
      .clone()
      .add(1, "weeks")
      .endOf(weekType)
```
The code also reads fine, if we add/subtract weeks from end, result is also a end